src/main/java/duke/DukeStorage.java

```diff
@@ -0,0 +1,127 @@
+package duke;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.util.Scanner;
+
+/**
+ * Class that defines the storage for the data needed for Duke
+ *
+ */
+public class DukeStorage {
+
+    /** The destination of the stored data */
+    public String path;
+
+    /**
+     * Constructor that initializes the storage for Duke
+     *
+     * @param path The destination of the stored data
+     */
+    public DukeStorage(String path) {
+        this.path = path;
+    }
+
+    /**
+     * Method to read tasks from storage
+     *
+     * @return List of tasks read from storage
+     * @throws DukeException Error thrown if file cannot be read
+     */
+    public TaskList readTasks() throws DukeException {
+        File file = new File(this.path);
+        if (file.exists()) {
+            try {
+                Scanner sc = new Scanner(file);
+                TaskList taskList = new TaskList();
+
+                String data = "";
+                while(sc.hasNext()) {
+                    data += sc.nextLine() + "\n";
+                }
+
+                if (data.equals("")){
+                    sc.close();
+                    return new TaskList();
+                }
+
+                String[] strArray = data.split("\n");
+
+                for (String str : strArray) {
+                    String[] parseTask = str.split(" \\| ");
+
+                    String taskType = parseTask[0];
+                    boolean isDone = parseTask[1].equals("1") ? true : false;
```

Comment:
Perhaps you could exclude the need for a ternary operation since equals() method already returns a boolean.
```suggestion
                    boolean isDone = parseTask[1].equals("1");
```